### PR TITLE
Updates to Multi-region RESTORE/BACKUP with updates in 21.2.3

### PR DESCRIPTION
**Pull request by kathancox**
Fixes [DOC-1872](https://cockroachlabs.atlassian.net/browse/DOC-1872), [DOC-1709](https://cockroachlabs.atlassian.net/browse/DOC-1709), [DOC-1158](https://cockroachlabs.atlassian.net/browse/DOC-1158), [DOC-723](https://cockroachlabs.atlassian.net/browse/DOC-723)

Removes limitations from 21.2: 

* backup can now backup up tables
* can now restore RBR

Also made adjustments to the Restore multi-region section to reflect these changes.

### CODE REVIEW

**Review by adityamaru — APPROVED** [review]

File: v21.2/backup.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ You can also backup:
 - [An individual database](#backup-a-database), which includes all of its tables and views.
 - [An individual table](#backup-a-table-or-view), which includes its indexes and views.
 
-    `BACKUP` only backs up entire tables; it _does not_ support backing up subsets of a table.
+    `BACKUP` only backs up entire tables; it **does not** support backing up subsets of a table.
```
Comment (livlobo) [inline v21.2/backup.md:22]: Could we add something that would indicate what a sub-set of a table could mean -- (e.g. you cannot backup a partition of a regional-by-row table). Although I don't know if "partition" is well-defined. I had to find out from the field team, that it means all rows in the RBR table that are in the same region.

File: v21.2/restore.md [hunk 2]
```diff
@@ -171,22 +171,20 @@ When a `RESTORE` fails or is canceled, partially restored data is properly clean
...
-
 * `RESTORE` supports restoring **non**-multi-region tables into a multi-region database and sets the table locality as [`REGIONAL BY TABLE`](multiregion-overview.html#regional-tables) to the primary region of the target database.
 
-* Restoring tables from multi-region databases with table localities set to `REGIONAL BY TABLE`, [`REGIONAL BY TABLE IN PRIMARY REGION`](set-locality.html#regional-by-table), and [`GLOBAL`](set-locality.html#global) to another multi-region database is supported.
+* Restoring tables from multi-region databases with table localities set to [`REGIONAL BY ROW`](multiregion-overview.html#regional-by-row-tables), `REGIONAL BY TABLE`, [`REGIONAL BY TABLE IN PRIMARY REGION`](set-locality.html#regional-by-table), and [`GLOBAL`](set-locality.html#global) to another multi-region database is supported.
 
 * When restoring a `REGIONAL BY TABLE IN PRIMARY REGION` table, if the primary region is different in the source database to the target database this will be implicitly changed on restore.
 
+* Restoring a [partition](partitioning.html) of a `REGIONAL BY ROW` table is not supported.
+
```
Comment (livlobo) [inline v21.2/restore.md:187]: Ah, I see you already have it! Ignore my previous comment.

**Review by livlobo — APPROVED** [review]
LGTM

**Review by taroface — APPROVED** [review]
LGTM with one formatting nit: I believe our style is to use `-` rather than `*` for unordered lists. I don't know that it makes a functional difference, but I've previously been asked to replace asterisks with hyphens for consistency.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12675#-:-MtKuO612Q4B4jUb869s:b-our3jz)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox and @livlobo)

**Comment by kathancox** [comment]
Thanks for the reminder on the * point @taroface!

**Pull request merged**